src/shared/Core/ICredentialStore.cs

```diff
@@ -28,14 +29,25 @@ public interface ICredentialStore
         /// <param name="service">Name of the service this credential is for. Use null to match all values.</param>
         /// <param name="account">Account associated with this credential. Use null to match all values.</param>
         /// <param name="secret">Secret value to store.</param>
+//        [Obsolete("Prefer AddOrUpdate(string, ICredential)")]
         void AddOrUpdate(string service, string account, string secret);
 
+        void AddOrUpdate(string service, ICredential credential)
+        => AddOrUpdate(service, credential.Account, credential.Password);
```

Comment:
Sorry that this issue is touching so many areas 😞. Probably best to switch to the old-school way in this file as well.